Cabal-tests/tests/UnitTests/Distribution/SPDX.hs

```diff
@@ -25,15 +25,7 @@ import Test.QuickCheck.Instances.Cabal ()
 spdxTests :: [TestTree]
 spdxTests =
     [ testProperty "LicenseId roundtrip" licenseIdRoundtrip
-#if MIN_VERSION_binary(0,7,0)
```

Comment:
These deleted test could be rather deactivated by placing them under
```c
#if false && MIN_VERSION_binary(0,7,0)
```
with a comment that they should be resurrected once the GHC is fixed and `Generic` instance of `LicenseId` is restored.

Reply:
@andreasabel According to every conversation I've had with GHC developers, including @bgamari, this is not a scenario that will arise before the heat death of the universe. :p

Reply:
Ok, well then maybe we can afford the entropy of removing these tests altogether...